### PR TITLE
Allow writing nil value using DynamoDB batch

### PR DESCRIPTION
**Pull request by aidan-kwon**
## Proposed changes

Currently, DynamoDB Batch doesn't allow writing `nil` value, but some logic tries to write `nil` value on the database. 
This PR allows that. 

Note: LevelDB and DynamoDB (non-batch) are able to write `nil` value. 

The logic writing `nil` value with a batch: https://github.com/klaytn/klaytn/blob/da66d4958a4706c4cdaced107c3d1b37eab9d22b/node/cn/bloombits.go#L143-L146

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by aidan-kwon** [comment]
The newly introduced test cases will be passed when https://github.com/klaytn/klaytn/pull/947 is merged and rebase on it.

File: storage/database/dynamodb_test.go [hunk 1]
```diff
@@ -91,6 +91,28 @@ func (s *SuiteDynamoDB) TestDynamoDB_Put() {
...
+	}
+	s.dynamoDBs = append(s.dynamoDBs, dynamo)
+
+	// write nil value
+	key := common.MakeRandomBytes(32)
+	s.Nil(dynamo.Put(key, nil))
+
+	// get nil value
+	ret, err := dynamo.Get(key)
+	s.Equal([]byte{}, ret)
```
Comment (ehnuje) [inline storage/database/dynamodb_test.go:107]: Shouldn't `ret` be `nil`?

**Review by ehnuje — COMMENTED** [review]
I think `*dynamoDB.get` should return `nil` instead of `[]byte{}` to distinguish empty value and nil value. How do you think?

**Comment by winnie-byun** [comment]
LGTM except for above comment

File: storage/database/dynamodb_test.go (hunk 1 above)
Comment (aidan-kwon) [reply storage/database/dynamodb_test.go:107]: @miloveme LevelDB `Get()` returns  `[]byte{}` for `nil` value. I changed MemoryDB `Get()` to adapt the same convention. Also, I applied the test case reading/writing `nil` value to all database types.

Comment (ehnuje) [reply storage/database/dynamodb_test.go:107]: Thanks for your explanation!

**Review by ehnuje — DISMISSED** [review]

**Comment by ehnuje** [comment]
Could you please check the test failure?
```
--- FAIL: TestDatabase_ParallelPutGet (1.75s)
panic: close of closed channel [recovered]
	panic: close of closed channel

goroutine 1289 [running]:
testing.tRunner.func1.1(0xea4f40, 0x112a830)
	/usr/local/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc0ee39f800)
	/usr/local/src/testing/testing.go:1075 +0x41a
panic(0xea4f40, 0x112a830)
	/usr/local/src/runtime/panic.go:969 +0x1b9
github.com/klaytn/klaytn/storage/database.(*dynamoDB).Close(...)
	/go/src/github.com/klaytn/klaytn/storage/database/dynamodb.go:401
github.com/klaytn/klaytn/storage/database.(*dynamoDB).deleteDB(0xc08c4a3720)
	/go/src/github.com/klaytn/klaytn/storage/database/dynamodb_test.go:353 +0x85
github.com/klaytn/klaytn/storage/database.newTestDynamoS3DB.func1()
	/go/src/github.com/klaytn/klaytn/storage/database/database_test.go:80 +0x2a
github.com/klaytn/klaytn/storage/database.TestDatabase_ParallelPutGet(0xc0ee39f800)
	/go/src/github.com/klaytn/klaytn/storage/database/database_test.go:107 +0xc8
testing.tRunner(0xc0ee39f800, 0x100b240)
	/usr/local/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
	/usr/local/src/testing/testing.go:1168 +0x2b3
FAIL	github.com/klaytn/klaytn/storage/database	3.995s
```

**Comment by aidan-kwon** [comment]
> Could you please check the test failure?

I fixed the failures. PTAL

**Review by ehnuje — DISMISSED** [review]

File: storage/database/database_test.go [hunk 2]
```diff
@@ -109,6 +134,61 @@ func TestDatabase_NotFoundErr(t *testing.T) {
...
+			val, err := db.Get(randStrBytes(100))
+			assert.Nil(t, val)
+			assert.Error(t, err)
+			assert.Equal(t, dataNotFoundErr, err)
+		}
+
+		// batch
+		{
+			batch := db.NewBatch()
+			fmt.Println(db.Type())
```
Comment (winnie-byun) [inline storage/database/database_test.go:167]:
```suggestion
```

**Review by winnie-byun — DISMISSED** [review]

**Review by winnie-byun — APPROVED** [review]

**Review by ehnuje — APPROVED** [review]

**Review by KimKyungup — APPROVED** [review]
